examples/taptree_of_horror/helper_fns.rs

```diff
@@ -0,0 +1,57 @@
+use std::str::FromStr;
+
+use bitcoin::bip32::{DerivationPath, Xpriv};
+use bitcoin::hashes::{ripemd160, sha256, Hash};
+use miniscript::descriptor::DescriptorSecretKey;
+use miniscript::ToPublicKey;
+use secp256k1::Secp256k1;
+
+pub fn produce_grim_hash(secret: &str) -> (sha256::Hash, ripemd160::Hash) {
+    let mut hash_holder = sha256::Hash::hash(secret.as_bytes());
+    for _i in 0..5 {
+        hash_holder = sha256::Hash::hash(hash_holder.as_byte_array());
+        //println!("{} hash: {}", i, hash_holder);
+    }
+
+    let ripemd_160_final = ripemd160::Hash::hash(hash_holder.as_byte_array());
+    (hash_holder, ripemd_160_final)
+}
+
+pub fn produce_kelly_hash(secret: &str) -> (sha256::Hash, sha256::Hash) {
+    let prepreimage: Vec<u8> = secret.as_bytes().to_vec();
+    let preimage_256_hash = sha256::Hash::hash(&prepreimage);
+    let result256_final = sha256::Hash::hash(&preimage_256_hash.to_byte_array());
+    (preimage_256_hash, result256_final)
+}
+
+pub fn produce_key_pairs(
+    desc: DescriptorSecretKey,
+    secp: &Secp256k1<secp256k1::All>,
+    derivation_without_index: &str,
+    _alias: &str,
+) -> (Vec<bitcoin::PublicKey>, Vec<Xpriv>) {
+    let mut pks = Vec::new();
+    let mut prvs = Vec::new();
+
+    let xprv = match &desc {
+        DescriptorSecretKey::XPrv(xpriv) => xpriv,
+        _ => panic!("not an xpriv"),
+    };
+
+    for i in 0..9 {
```

Comment:
In 1282d30653a3cb4e188e59d5a3dd8de909aa9477:

I would suggest lifting this to a constant `KEYS_PER_CHARACTER` or something.

Reply:
Went with `KEYS_PER_PERSONA`